extensions/wikia/DesignSystem/DesignSystemHelper.class.php

```diff
@@ -59,4 +67,28 @@ public static function getCachedSvg( $name ) {
 
 		return $xml;
 	}
+
+	public static function renderText( $fields ) {
+		if ( $fields['type'] === 'text' ) {
+			return $fields['value'];
+		}
+
+		if ( $fields['type'] === 'translatable-text' ) {
+			if ( isset( $fields['params'] ) ) {
+				$paramsRendered = [];
+
+				foreach ( self::MESSAGE_PARAMS_ORDER[$fields['key']] as $index => $paramKey ) {
+					$paramsRendered[] = self::renderText( $fields['params'][$paramKey] );
+				}
+
+				return wfMessage( $fields['key'] )->rawParams( $paramsRendered )->escaped();
+			} else {
+				return wfMessage( $fields['key'] )->escaped();
+			}
+		}
+
+		if ( $fields['type'] === 'link-text' ) {
+			return '<a href="' . $fields['href'] . '">' . self::renderText( $fields['title'] ) . '</a>';
```

Comment:
The href isn't escaped here and if the title field is text it isn't escaped either (translatable text is though). I'd suggest building the link with `Html::rawElement` which will escape the attributes passed to it. It won't escape the content of the link (`Html::element` would), but since you need to escape the other return options for `renderText`, you can use `rawElement` and ensure all other text returned by `renderText` is escaped.